walkthrough/05-aws/01-no-stacks/README.md

```diff
@@ -0,0 +1,180 @@
+# 01 - No Stacks
+
+To set a baseline, this walkthrough will set up a Stack without the use of any `terragrunt.stack.hcl` files.
+
+This will motivate the need for `terragrunt.stack.hcl` files in the next walkthrough.
+
+## Architecture
+
+```bash
+$ tree
+.
+├── README.md
+├── _shared
+│   ├── api.hcl
+│   └── db.hcl
+├── live
+│   ├── dev
+│   │   ├── environment.hcl
+│   │   ├── services
+│   │   │   └── api
+│   │   │       └── terragrunt.hcl
+│   │   └── storage
+│   │       └── db
+│   │           └── terragrunt.hcl
+│   └── prod
+│       ├── environment.hcl
+│       ├── services
+│       │   └── api
+│       │       └── terragrunt.hcl
+│       └── storage
+│           └── db
+│               └── terragrunt.hcl
+├── modules
+│   ├── api
+│   │   ├── logs.tf
+│   │   ├── main.tf
+│   │   ├── outputs.tf
+│   │   ├── role.tf
+│   │   ├── variables.tf
+│   │   └── versions.tf
+│   └── db
+│       ├── main.tf
+│       ├── outputs.tf
+│       ├── variables.tf
+│       └── versions.tf
+├── scripts
+│   ├── id.sh
+│   └── package.sh
+├── src
+│   └── index.mjs
+└── terragrunt.hcl
+```
+
+The resources that are going to be provisioned as part of this walkthrough are fairly simple:
+
+1. A DynamoDB table that will store a count.
+2. A Lambda function that will accept `GET` and `POST` requests to read and increment a count in the DynamoDB table.
+
+The two will be provisioned with their own OpenTofu state, and will integrate with each other via a `dependency` block on the `api` Unit.
+
+Furthermore, the "Stack" of resources will be replicated across two environments: `dev` and `prod`.
+
+To reduce repetition, the OpenTofu modules the Units will use have been abstracted into the `modules` directory, and the shared configuration for the Units has been abstracted into the `_shared` directory.
+
+In addition, some helper scripts have been added in a `scripts` directory to package the source code in `src` for the Lambda function, and a `id.sh` script to generate a unique ID for the project.
+
+The `id.sh` script is useful for this example, as it will generate a unique ID that can be used to ensure that the resources created in the walkthrough are globally unique.
+
+This was added so that anyone following along with this walkthrough can run the code without having to worry about naming conflicts from S3 buckets provisioned by others.
+
+### Note on Configuration Lifecycles
+
+Many IaC authors starting out writing IaC see fully formed configurations like this and can get intimidated by the amount of configuration they are looking at.
+
+It's important to remember that most IaC configurations are authored incrementally, and that the configuration lifecycle is iterative.
+
+When writing up this walkthrough, I did so in the following order:
+
+1. I created the `live/dev/services/api/terragrunt.hcl` file and created a `live/dev/services/api/main.tf` file with the minimal configuration to create a Lambda function.
```

Comment:
and created a `modules/api/main.tf`

Reply:
It's how we teach users Terragrunt:
https://terragrunt.gruntwork.io/docs/getting-started/quick-start/

It's the "minimal overhead" approach to adopting Terragrunt if you're currently using OpenTofu.

Reply:
Understood. In my case, I was confused why you started that way. Might be worth clarifying.